accept/src/botReview.ts

```diff
@@ -2,10 +2,67 @@ import { getInput, info, warning } from '@actions/core'
 import { getOctokit } from '@actions/github'
 import { getPR, getShaFromBranch } from '../../lib/github'
 import { detectIssueKey } from './detectIssueKey'
+import { getEventData } from '../../lib/event'
 
 // id of exivity bot
 const EXIVITY_BOT = 53756225
 
+interface IDObject {
+  id: number
+}
+
+// Checks if all check runs connected to this ref have completed successfully
+async function areChecksDone(
+  octokit: ReturnType<typeof getOctokit>,
+  ref: string,
+  repo: string
+): Promise<boolean> {
+  const checkResult = await octokit.checks.listForRef({
+    owner: 'exivity',
+    repo,
+    ref,
+  })
+
+  return checkResult.data.check_runs?.every(
+    (check) => check.status === 'completed' && check.conclusion === 'success'
+  )
+}
+
+// Checks if the branch that had the event triggering this action is ready for scaffold to run,
+// or that we need to wait for a next event.
+async function checkIfReady(
+  octokit: ReturnType<typeof getOctokit>,
+  ref: string,
+  repo: string
+): Promise<boolean> {
+  const event = process.env['GITHUB_EVENT_NAME']
+
+  const eventData = await getEventData()
+
+  if (
+    event === 'check_run' &&
+    eventData.check_run?.check_suite.pull_requests.requested_reviewers.some(
+      (reviewer: IDObject) => reviewer.id === EXIVITY_BOT
+    )
+  ) {
+    return await areChecksDone(octokit, ref, repo)
+  }
+
+  if (event === 'pull_request') {
+    let reviewers = eventData.requested_reviewer
+
+    if (!Array.isArray(reviewers)) {
+      reviewers = [reviewers]
+    }
+
+    if (reviewers.some((reviewer: IDObject) => reviewer.id === EXIVITY_BOT)) {
+      return await areChecksDone(octokit, ref, repo)
```

Comment:
nit: unnecessary await
```suggestion
      return areChecksDone(octokit, ref, repo)
```